dotnet/src/IntegrationTests/Connectors/Memory/Milvus/MilvusMemoryStoreTests.cs

```diff
@@ -220,6 +220,47 @@ public async Task GetNearestMatchesAsync(bool withEmbeddings)
             });
     }
 
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false)]
+    public async Task GetNearestMatchesWithMetricTypeAsync(bool withEmbeddings)
+    {
+        //Create collection with default, Ip metric
+        await this.Store.CreateCollectionAsync(CollectionName);
+        await this.InsertSampleDataAsync();
+        // There seems to be some race condition where the upserted data above isn't taken into account in the search below and zero results are returned...
+        await Task.Delay(1000);
+
+        //Search with Ip metric, run correctly
+        List<(MemoryRecord Record, double SimilarityScore)> ipResults =
+            this.Store.GetNearestMatchesAsync(CollectionName, new[] { 5f, 6f, 7f, 8f, 9f }, limit: 2, withEmbeddings: withEmbeddings).ToEnumerable().ToList();
+
+        Assert.All(ipResults, t => Assert.True(t.SimilarityScore > 0));
+
+        //Set the store to Cosine metric, without recreate collection
+        this.Store = new(this._milvusFixture.Host, vectorSize: 5, port: this._milvusFixture.Port, metricType: SimilarityMetricType.Cosine, consistencyLevel: ConsistencyLevel.Strong);
+
+        //An exception willl be thrown here, the exception message includes "metric type not match"
+        MilvusException milvusException = Assert.Throws<MilvusException>(() => this.Store.GetNearestMatchesAsync(CollectionName, new[] { 5f, 6f, 7f, 8f, 9f }, limit: 2, withEmbeddings: withEmbeddings).ToEnumerable().ToList());
+
+        Assert.NotNull(milvusException);
+
+        Assert.Contains("metric type not match", milvusException.Message);
+
+        //Recreate collection with Cosine metric
+        await this.Store.DeleteCollectionAsync(CollectionName);
+        await this.Store.CreateCollectionAsync(CollectionName);
+        await this.InsertSampleDataAsync();
+        // There seems to be some race condition where the upserted data above isn't taken into account in the search below and zero results are returned...
+        await Task.Delay(1000);
```

Comment:
If there is a race condition, the test may periodically fail during our integration test run. I'm wondering if new change can be verified with unit testing instead of integration testing by mocking Milvus client?

Reply:
This test was written based on the current use of `MilvusContainer`. This issue is also present in other existing unit tests.

![image](https://github.com/microsoft/semantic-kernel/assets/17045801/d3780cac-c432-4623-ada8-e8438d7dacc5)

I have modified the newly added unit tests to avoid race issues.